modules/raft/src/main/java/org/apache/ignite/raft/jraft/core/NodeImpl.java

```diff
@@ -1033,6 +1034,33 @@ public boolean init(final NodeOptions opts) {
         // Adds metric registry to RPC service.
         this.options.setMetricRegistry(this.metrics.getMetricRegistry());
 
+        //Wait committed
+        long commitIdx = logManager.getLastLogIndex();
+
+        if (commitIdx > fsmCaller.getLastAppliedIndex()) {
+            CountDownLatch applyCommitLatch = new CountDownLatch(1);
+
+            LastAppliedLogIndexListener lnsr = lastAppliedLogIndex -> {
+                if (lastAppliedLogIndex >= commitIdx) {
+                    applyCommitLatch.countDown();
+                }
+            };
+
+            fsmCaller.addLastAppliedLogIndexListener(lnsr);
+
+            fsmCaller.onCommitted(commitIdx);
+
+            try {
+                applyCommitLatch.await();
```

Comment:
How long would it take? Did it affect the duration of current tests?
I hope we start these nodes in parallel at least.

Reply:
I hope this change will not have significant effect, but I understand the pure RAFT test become longer.
We can do this feature is configurable if it will be required in the future.